### PR TITLE
Added JWT support

### PR DESCRIPTION
**Pull request by ynik404**

### CODE REVIEW

File: src/main/java/mate/academy/security/JwtAuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/security/JwtAuthenticationFilter.java:46]: Create constant for "Bearer "

File: src/main/java/mate/academy/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            return bearerToken.substring(7);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/security/JwtAuthenticationFilter.java:47]:
```suggestion
            return bearerToken.substring(constant.length());
```

File: src/main/java/mate/academy/security/JwtUtil.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import java.util.function.Function;
+import javax.crypto.SecretKey;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class JwtUtil {
+    private final SecretKey secret;
+
+    @Value("${jwt.expiration}")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/security/JwtUtil.java:19]: Didn't find application.properties with jwt.expiration an jwt.secret in this PR

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by MentorMateAcademy — APPROVED** [review]

**Pull request merged**